### PR TITLE
Port rust binding from Unblob

### PR DESCRIPTION
**Pull request by vlaci**
- Code is from https://github.com/onekey-sec/unblob/tree/main/rust
- Initialized packaging
- Pyright specific changes
- Pre-commit hooks
- CI workflows
- README with getting started with development information

### CODE REVIEW

**Comment by qkaiser** [comment]
That's a lot of infra for a single function exposed from Rust :) I hope we'll move more things to Rust to make use of it.

I don't see any obvious issues since it's a copy paste with a few adjustments. Should we expect a matching MR in unblob that removes https://github.com/onekey-sec/unblob/tree/main/rust ?

**Comment by vlaci** [comment]
> That's a lot of infra for a single function exposed from Rust :) I hope we'll move more things to Rust to make use of it.
> 
> I don't see any obvious issues since it's a copy paste with a few adjustments. Should we expect a matching MR in unblob that removes https://github.com/onekey-sec/unblob/tree/main/rust ?

I wanted it to be as clean as possible. Even if we never touch this again, the better to keep it in a tidy state :)

Yep, if it reaches PyPI, we can remove that one, and make Unblob a pure package without architecture  constrants

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            };
+          };
+      in
+      {
+        checks = unblob-native.tests // {
+          # Build the crate as part of `nix flake check` for convenience
+          inherit libunblob-native;
+
+          # Run clippy (and deny all warnings) on the crate source,
+          # again, resuing the dependency artifacts from above.
```
Comment (e3krisztian) [inline flake.nix]: reUSing

**Review by e3krisztian — APPROVED** [review]

File: flake.nix (hunk 1 above)
Comment (vlaci) [reply flake.nix]: https://github.com/ipetkov/crane/blob/8708b19627b2dfc2d1ac332b74383b8abdd429f0/examples/quick-start/flake.nix#L76 :sweat_smile:

**Pull request merged**